File: implementations/node/sdk/src/contracts/replay/execution-replay-request.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";

export interface AiSdkExecutionReplayRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.executionReplayRequest;
  readonly strictDeterminism?: boolean;
  readonly includeDiagnostics?: boolean;
  readonly reason?: string;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/replay/execution_replay_request.py
from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkExecutionReplayRequest(AiSdkWireModel):
    schema_version: int = AiSdkSchemaVersions.EXECUTION_REPLAY_REQUEST
    strict_determinism: bool = True
    include_diagnostics: bool = True
    reason: str | None = None
    correlation_id: str | None = None
